src/foam/nanos/bench/BenchmarkRunner.java

```diff
@@ -111,17 +128,24 @@ public void execute(final X x) {
     long startTime = System.currentTimeMillis();
 
     // execute all the threads
-    for ( Thread thread : threads ) {
-      thread = new Thread(new Runnable() {
+    for ( int i = 0 ; i < threadCount_ ; i++ ) {
+      final int tno = i;
+      Thread thread = new Thread(group, new Runnable() {
         @Override
         public void run() {
-          for ( int i = 0 ; i < invocationCount_ ; i ++ ) {
+          for ( int j = 0 ; j < invocationCount_ ; j++ ) {
             test_.execute(x);
+            count.incrementAndGet();
```

Comment:
This seems like it will add unnecessary contention between the threads.  Now they all have to synchronize on a single integer just so we can count how many were completed.  Can't we just calculate that upfront?  We know its going to be invocationCount_ * threadCount_ unless one of them throws an error and doesn't complete all their work.

Reply:
On the thought of throwing errors, maybe we should try/catch test_.execute and abort the whole benchmark if something goes wrong.

Reply:
Yes, a heavily contested lock like this is really going to throw off the numbers.